atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/metrics/TargetedSweepMetrics.java

```diff
@@ -135,7 +140,16 @@ private static Long estimateMillisSinceTs(Long timestamp, Clock clock, Function<
             if (timestamp == null) {
                 return null;
             }
-            return clock.getTimeMillis() - tsToMillis.apply(timestamp);
+            long timeBeforeRecomputing = System.currentTimeMillis();
+            long result = clock.getTimeMillis() - tsToMillis.apply(timestamp);
+
+            long timeTaken = System.currentTimeMillis() - timeBeforeRecomputing;
+            if (timeTaken > TimeUnit.SECONDS.toMillis(10)) {
```

Comment:
if this is to let us know if it should be async or not, should we drop them by a factor of 10?

Reply:
If we ever get this, we should probably drop to one day or even 12 hours, yes. The numbers chosen were arbitrary, but as long as it takes less than a second I think we are fine